components/ui-api-layer/internal/domain/k8s/resourcequotastatus_service.go

```diff
@@ -106,45 +114,66 @@ func (svc *resourceQuotaStatusService) CheckResourceQuotaStatus(environment stri
+			exceeded := svc.checkAvailableResources(replicaUsage, resourcesToCheck, resourceQuotas)
+			for _, exc := range exceeded {
+				result = svc.ensureStatusesMapEntry(exc.rqName, exc.resourceName, result)
+				result[exc.rqName][exc.resourceName] = append(result[exc.rqName][exc.resourceName], svc.affectedResourceRef(ssKind, ss.Name))
 			}
 		}
 	}
-	return gqlschema.ResourceQuotaStatus{Exceeded: false}, nil
+	return result, nil
 }
 
-func (svc *resourceQuotaStatusService) checkPodsUsage(environment string, labelSelector map[string]string, resourcesToCheck []v1.ResourceName, resourceQuotas []*v1.ResourceQuota) (bool, error) {
-	pods, err := svc.podLister.ListPods(environment, labelSelector)
+func (svc *resourceQuotaStatusService) getDefaultLimits(environment string, resourcesToCheck []v1.ResourceName) (map[v1.ResourceName]*resource.Quantity, error) {
+	limitRanges, err := svc.lrLister.List(environment)
 	if err != nil {
-		return false, errors.Wrapf(err, "while listing pods [environment: %s][labelSelector: %v]", environment, labelSelector)
-	}
-
-	if len(pods) > 0 {
-		// We are checking only one Replica from the given Set, because every replica has the same resources usage.
-		limits := svc.containersUsage(pods[0].Spec.Containers, resourcesToCheck)
-		for _, rq := range resourceQuotas {
-			for _, name := range resourcesToCheck {
-				hard, hardExists := rq.Spec.Hard[name]
-				used := rq.Status.Used[name]
-				if hardExists && hard.Value()-used.Value() < limits[name].Value() {
-					return true, nil
+		return nil, errors.Wrapf(err, "while listing limitRanges [environment: %s]", environment)
+	}
+	limits := make(map[v1.ResourceName]*resource.Quantity)
+	for _, name := range resourcesToCheck {
+		limits[name] = &resource.Quantity{}
+	}
+	for _, lr := range limitRanges {
+		for _, limit := range lr.Spec.Limits {
+			if limit.Type == v1.LimitTypeContainer {
+				if limit.DefaultRequest.Memory().Value() > limits[v1.ResourceRequestsMemory].Value() {
+					limits[v1.ResourceRequestsMemory] = limit.DefaultRequest.Memory()
+				}
+				if limit.Default.Memory().Value() > limits[v1.ResourceLimitsMemory].Value() {
+					limits[v1.ResourceLimitsMemory] = limit.Default.Memory()
+				}
+				if limit.Default.Cpu().Value() > limits[v1.ResourceLimitsCPU].Value() {
+					limits[v1.ResourceLimitsCPU] = limit.Default.Cpu()
 				}
+				if limit.DefaultRequest.Cpu().Value() > limits[v1.ResourceRequestsCPU].Value() {
+					limits[v1.ResourceRequestsCPU] = limit.DefaultRequest.Cpu()
+				}
+			}
+		}
+	}
+	return limits, nil
+}
+
+func (*resourceQuotaStatusService) checkAvailableResources(limits map[v1.ResourceName]*resource.Quantity, resourcesToCheck []v1.ResourceName, resourceQuotas []*v1.ResourceQuota) []exceededRef {
+	result := make([]exceededRef, 0)
+	for _, rq := range resourceQuotas {
+		for _, name := range resourcesToCheck {
+			hard, hardExists := rq.Spec.Hard[name]
+			used := rq.Status.Used[name]
```

Comment:
Scenario:
1. User creates the ResourceQuota **ABC**
2. In etcd ResourceQuota **ABC** is available and have the `hard` object set under `spec` field
3. Controller for some reason cannot handle created **ABC**  ResourceQuota, so controller will not populate the `Used` section


When sth like that will take place then you will have probably panic in `if` statement